chapters/cls-metrics.qmd

```diff
@@ -0,0 +1,1693 @@
+
+We'll fit the model to the training set and, for the most part, use the validation set to demonstrate the various metrics. @fig-naive-bayes-probs shows the distribution of the probability of the event (poisonous) for the validation set. 
+
+```{r}
+#| label: fig-naive-bayes-probs
+#| echo: false
+#| out-width: 70%
+#| fig-width: 6
+#| fig-height: 4.25
+#| fig-cap: The distribution of the validation set class probability estimates for a naive Bayes model, split by the true class of the mushroom. 
+
+nb_val_pred %>% 
+  ggplot(aes(.pred_poisonous)) + 
+  geom_vline(xintercept = 1 / 2, col = "blue", lty = 3) +
+  geom_histogram(col = "white", binwidth = 0.02) + 
+  facet_wrap(~ class, ncol = 1, labeller = "label_both") + 
+  labs(x = "Probability of Being Poisonous")
+```
+
+These distributions demonstrate a few attractive characteristics. First, the probability estimates span the entire range of possible values, and each distribution peaks at appropriately extreme values (i.e., zero or one). Conversely, the model rarely produces "confidently incorrect" predictions (i.e., erroneously predicting with a large probability estimate). We'll see this model’s calibration and separation properties in later sections.   
+
+## Assessing Hard Class Predictions {#sec-cls-hard-metrics}
+
+We’ll start by focusing on the qualitative predictions: `"poisonous"` or `"edible"`. Unless otherwise stated, hard class predictions for two-class problems are assigned in this chapter using a 50{{< pct >}} cutoff. Alternate cutoffs are investigated in @sec-roc.
+
+One of the most fundamental ways to inspect and understand how well hard class prediction performs is the **confusion matrix**, simply a cross-tabulation of the observed and predicted classes. From this, we can conceptualize where the model underperforms. An excellent model has the largest counts along the diagonal where the observed and predicted classes are the same. For the naive Bayes model, @tbl-confusion-matrix shows the confusion matrix for the mushroom data’s validation set. 
+
+```{r}
+#| label: cls-metrics
+#| include: false
+
+nb_conf_mat <-
+  nb_val_pred %>% 
+  conf_mat(class, estimate = .pred_class)
+
+conf_mat_table <- 
+  as_tibble(as.matrix(nb_conf_mat$table)) %>% 
+  pivot_wider(id_cols = c(Prediction), names_from = c(Truth), values_from = n) %>% 
+  mutate(Prediction = gsub("class_", "Class ", Prediction)) %>% 
+  mutate(blank = "  ") %>% 
+  relocate(blank, .after = "Prediction") 
+
+acc_int <- binom.test(sum(diag(nb_conf_mat$table)), sum(nb_conf_mat$table), conf.level = 0.9)
+```
+
+::::::: columns
+::: {.column width="37%"}
+:::
+
+:::: {.column width="26%"}
+::: {#tbl-confusion-matrix}
+```{r}
+#| label: confusion-matrix
```

Comment:
For the latest version of Quarto, when we have the "fenced chunks" like this (with the `:::` parts), the `::: {#tbl-confusion-matrix}` represents the cross-reference and if we use the `tbl-` prefix in the `#| label:` part, it goofs things up. 

It's also true for figures. 

This isn't well documented.